File: 300.500/455.AssignCookies.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class AssignCookies
{
public:
    int findContentChildren(vector<int> g, vector<int> s) {
        sort(g.begin(), g.end(), [](int a, int b){
            return a < b;
        });
        sort(s.begin(), s.end(), [](int a, int b){
            return a < b;
        });
        int i = 0, j = 0, r = 0;
        while(i < s.size() && j < g.size()){
            if(s[i] >= g[j]){
                i++;
                j++;
                r++;
            }else{
                i++;
            }
        }
        return r;
    }
};

int main(int argc, char const *argv[])
{
    AssignCookies cls;
    cout << cls.findContentChildren(vector<int>{1,2,3}, vector<int>{1,1}) << " -> 1" << endl;
    cout << cls.findContentChildren(vector<int>{1,2,3}, vector<int>{1,2}) << " -> 2" << endl;
    cout << cls.findContentChildren(vector<int>{10,9,8,7}, vector<int>{5,6,7,8}) << " -> 2" << endl;
    return 0;
}

File: go/300.500/455.AssignCookies.go
package main

import (
	"fmt"
	"sort"
)

func findContentChildren(g []int, s []int) int {
	sort.Slice(g, func(i, j int) bool {
		return g[i] < g[j]
	})
	sort.Slice(s, func(i, j int) bool {
		return s[i] < s[j]
	})
	i, j, r := 0, 0, 0
	for i < len(g) && j < len(s) {
		if s[j] >= g[i] {
			i++
			j++
			r++
		} else {
			j++
		}
	}
	return r
}

func main() {
	fmt.Println(findContentChildren([]int{10, 9, 8, 7}, []int{5, 6, 7, 8}), 2)
	fmt.Println(findContentChildren([]int{1, 2, 3, 4}, []int{1, 2}), 2)
}
